src/main/java/racingcar/Cars.java

```diff
@@ -1,29 +1,58 @@
 package racingcar;
 
+import racingcar.strategy.MovingStrategy;
+
 import java.util.ArrayList;
 import java.util.List;
 
 public class Cars {
 
     private final List<Car> cars;
 
-    public Cars(int carCount) {
-        cars = createCars(carCount);
+    public Cars(String[] carNames) {
+        this.cars = createCars(carNames);
+    }
+
+    public Cars(List<Car> cars) {
+        this.cars = cars;
     }
 
-    private List<Car> createCars(int carCount) {
+    private List<Car> createCars(String[] carNames) {
         final List<Car> cars = new ArrayList<>();
-        for (int i = 0; i < carCount; i++) {
-            cars.add(new Car());
+        for (String carName : carNames) {
+            cars.add(new Car(carName));
+        }
+        return cars;
```

Comment:
스트림을 이용하면 이렇게도 쓸 수 있을거 같아요. 
```suggestion
return Arrays.stream(carNames) 
        .map(Car::new) 
        .collect(Collectors.toList());        
```